tensorflow/python/keras/layers/convolutional.py

```diff
@@ -230,6 +230,15 @@ def build(self, input_shape):
     tf_op_name = self.__class__.__name__
     if tf_op_name == 'Conv1D':
       tf_op_name = 'conv1d'  # Backwards compat.
+    
+    # Check if output shapes are valid
+    # They must not have 0 entries along any dimension
+    output_shapes = self.compute_output_shape(input_shape).as_list()
+    output_shapes = list(filter((None).__ne__, output_shapes))
+    if not all(output_shapes):
```

Comment:
all() operator here is a bit unclear, please change to 0 in output_shapes, or do specific checks on the dimension.